modules/git/repo_tag.go

```diff
@@ -161,37 +162,83 @@ func (repo *Repository) GetTag(name string) (*Tag, error) {
 
 // GetTagInfos returns all tag infos of the repository.
 func (repo *Repository) GetTagInfos(page, pageSize int) ([]*Tag, int, error) {
-	// TODO this a slow implementation, makes one git command per tag
-	stdout, err := NewCommand(repo.Ctx, "tag").RunInDir(repo.Path)
+	forEachRefFmt := foreachref.NewFormat("objecttype", "refname:short", "object", "objectname", "creator", "contents", "contents:signature")
+
+	stdout, err := NewCommand(repo.Ctx, "for-each-ref", "--format", forEachRefFmt.Flag(), "--sort", "-*creatordate", "refs/tags").RunInDir(repo.Path)
```

Comment:
If the `stdout` maybe very large, we can use `RunWithContext` and use a `io.Pipe` to create a reader for `Parser` and a writer for `Git`.

Reply:
Fixed in 152a12d9b